mithril-stm/src/stm.rs

```diff
@@ -304,13 +311,40 @@ impl StmInitializer {
         if my_index.is_none() {
             return Err(RegisterError::UnregisteredInitializer);
         }
+        Ok(my_index)
+    }
 
-        Ok(StmSigner {
-            mt_index: my_index.unwrap(),
+    /// Returns the StmSigner of the initializer for given index.
+    pub fn new_signer(self, index: Index) -> StmSigner {
```

Comment:
The only way to provide an index is via a closed registration. However, we don't want to make a requirement to initialise an StmSigner to have a closed registration, as that might not happen in the full node case. 

Actually, we don't need to create an StmSigner from an StmInitializer. The reason why we have that for the `StmSignerAvk` is because we want to enforce the creation of the latter only if a closed registration exists (which is run by the initializer). However, we don't need that for the StmSigner, and therefore, we could directly have a `new()` function associated to `StmSigner` that takes as input the same values as the `setup` function above, plus the list of all eligible parties (which it already knows as it is running a full node).